File: designPattern/Singleton/src/Singleton.java
public class Singleton {
    private static Singleton singleton = new Singleton();
    //다른 곳에서 new Singleton()을
    //즉 인스턴스를 못만들게 하기 위해서
    //private로 선언 했다.
    private Singleton(){
        System.out.println("making Singleton ");
    }
    //static 메소드를 만들어 줌으로 써
    //클래스명.메소드이름()으로 호출 가능하다.
    static Singleton getInstance(){
        return singleton;
    }
}

File: designPattern/Singleton/javascript/Singleton.js
let instance;
class Singleton{

	constructor(){

		if(instance) 

			return instance;

		instance = this;

//이 클래스의 객체를 만들어서 넣어준다.

	}

}



let A = new Singleton();

let B = new Singleton();



if(A === B)

console.log("같다");

else 

console.log("다르다");

